### DISCUSSION TITLE
Initializing the _hist material property for the phase field fracture model from the input file

### DISCUSSION THREAD

**Discussion by Aoiosamas** [idaholab/moose#16991] [Q&A Modules: Solid mechanics] [answered]
Hello all,

I have a question regarding the history variable approach used in the phase field fracture model in MOOSE, when I have a distribution of preexisting cracks. The question is about the initialization of the _hist material property. According to this post (https://groups.google.com/g/moose-users/c/Bf_nkvs-j3k/m/yXib3yPMBwAJ), I have to hard code ComputeIsotropicLinearElasticPFFractureStress::initQpStatefulProperties() function, if I'm not interested in creating the cracks in the mesh itself (real cracks). I was wondering if it has become possible somehow to initialize this history variable in the input time, e.g., using an Auxillary variable or a material property. Any insights/suggestions are really appreciated.

Thank you,
Ahmed

**Comment by hugary1995** [comment, marked as answer]
Do you have to use a history field? I recommend against it. Instead, it's better to apply initial condition on c, and enforce the irreversibility using `VariableOldValueBoundsAux`.

Now, if you insist on using history to initialize the crack, you can take a look at this PR: [https://github.com/idaholab/moose/pull/15102](https://github.com/idaholab/moose/pull/15102) which tried to do what you are describing here. It seems to me that PR was really close to be merged, I don't know why it got closed in the end. Like I said there, I'm not a big fan of the history approach, but I also have no objection in us supporting that capability.

**Reply by Aoiosamas** [reply]
This is very helpful. Thank you so much.
